docs/openshift/deploy.md

```diff
@@ -117,3 +117,53 @@ Run `acs-engine generate examples/openshift.json`
 After couple of minutes, your OpenShift web console should be accessible at `https://${dnsprefix}.${location}.cloudapp.azure.com:8443/`. You can log in with `clusterUsername` and `clusterPassword` values you set in the `openshift.json` file or with your Azure account if you've used AAD integration.
 
 For next steps, see [getting started documentation](https://docs.openshift.org/latest/getting_started/index.html) on OpenShift website.
+
+## Custom VNET
+
+ACS Engine supports deploying into an existing VNET. Operators must specify the ARM path/id of Subnets for the `masterProfile` and  any `agentPoolProfiles`, as well as the first IP address to use for IP static IP allocation in `firstConsecutiveStaticIP`. Additionally, to prevent source address NAT'ing within the VNET, we assign to the `vnetCidr` property in `masterProfile` the CIDR block that represents the usable address space in the existing VNET.
+
+To create a vnet and a subnet, for example:
+
+```bash
+az network vnet create -g $RESOURCE_GROUP -n $VNET_NAME --address-prefixes 10.239.0.0/16 --subnet-name $SUBNET_NAME --subnet-prefix 10.239.0.0/24
+```
+To get the `vnetSubnetId`:
+
+```bash
+az network vnet subnet show -n $SUBNET_NAME -g $RESOURCE_GROUP --vnet-name $VNET_NAME | jq -r .id
+```
+
+Edit the [OpenShift with custom vnet cluster definition](/examples/vnet/openshift-vnet.json) and fill out the required values (every value with empty default `""` must be filled in).
+
+To avoid IP address allocation collision during deployment:
```

Comment:
Have you actually seen this behaviour happening? Does Azure DHCP really not guarantee not to issue reserved static IPs to dynamic clients?

Reply:
Ah, I see.  The agent nics must be created after the master nic.  Apply this diff to parts/k8s/kubernetesagentresourcesvmas.t, then remove the comments here about collisions.  You could optionally mention https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-faq#are-there-any-restrictions-on-using-ip-addresses-within-these-subnets if you want - I don't mind either way.

```diff
diff --git a/parts/k8s/kubernetesagentresourcesvmas.t b/parts/k8s/kubernetesagentresourcesvmas.t
index 06cb24b2..59f1c910 100644
--- a/parts/k8s/kubernetesagentresourcesvmas.t
+++ b/parts/k8s/kubernetesagentresourcesvmas.t
@@ -12,7 +12,9 @@
       "[variables('vnetID')]"
 {{end}}
 {{else}}
-{{if not .IsCustomVNET}}
+{{if .IsCustomVNET}}
+      "[concat(variables('masterVMNamePrefix'), 'nic-0')]",
+{{else}}
       "[variables('vnetID')]",
 {{end}}
 {{if eq .Role "infra"}}
```

Reply:
There is currently a validation that checks for `MasterProfile.FirstConsecutiveStaticIP`. See https://github.com/Azure/acs-engine/blob/master/pkg/api/vlabs/validate.go#L1147-L1149